src/monitor/main_monitor.cpp

```diff
@@ -64,6 +64,16 @@ MainMonitor::MainMonitor() : trace_(), metrics_(trace_)
         }
     }
 
+    if (config().use_block_io)
+    {
+        matcher_ = std::make_unique<perf::bio::Matcher>(trace_);
+        for (const auto& cpu : Topology::instance().cpus())
+        {
+            bio_monitors_.emplace_back(std::make_unique<BioMonitor>(trace_, Cpu(cpu.id), matcher_));
```

Comment:
Since C++17, emplace_back returns a `reference`, I'd rather use that to `start()`